### DISCUSSION TITLE
Connection Occassionally Dropping and Re-Connecting

### DISCUSSION THREAD

**Discussion by nkreiger** [supabase/supabase#13562] [Questions] [answered]
Is it normal for a connection (via superbaseclient-js) to drop and reconnect occassionally? Does that mean I could potentially miss events during high traffic periods? I see no errors in the supabase logs 

- I'm running a 3 pod instance with auto-scaling on a kubernetes cluster). Using the bitnami supabase realtime image, with pieced together helm charts from that same repository.

**Comment by GaryAustin1** [comment]
Yes, the connection can drop for lots of reasons and you will lose data during the automatic re-connections.
You basically need to restart the connection a new and load your initial data after a drop if you don't want to miss anything.

This is a bit dated... but covers some of what goes on... https://github.com/orgs/supabase/discussions/5641

**Reply by nkreiger** [reply]
Hi @GaryAustin1 , I think I found that I'm actually running into an issue:

```
16:17:16.385 project=realtime external_id=realtime [error] Error polling replication: %DBConnection.ConnectionError{
  message: "tcp recv: closed",
  severity: :error,
  reason: :error
}
```

Even if I stop and close the channel, I continuously see this until I completly kill and restart realtime.

after a certain amount of time

**Reply by GaryAustin1** [reply]
Yeah, that does not sound good.  
What does it mean to "kill and restart realtime"?
Any errors in the dashboard realtime logs?
If you think it is an issue then either generate an issue at github supabase/realtime or a support request might be in order.

**Reply by nkreiger** [reply]
@GaryAustin1 I'm running the realtime service self-hosted in kubernetes, and I have to kill it (restart the pods). Can't seem to figure out what might be causing the drop, I can open an issue. 

The first error is:

"tcp recv (idle): closed",

Which indicates it could be getting closed by Cloud SQL (GCP). But the timeout is not set, and I'm not sure why Realtime wouldn't try to reconnect?

**Reply by GaryAustin1** [reply]
Yeah you definitely want to at least change the title or ask a new question / issue as you will need someone else to help if you don't solve it.  I'm not familiar enough with realtime in a local setting, let alone what you are doing.

**Reply by nkreiger** [reply, marked as answer]
@GaryAustin1 for knowledge share, I believe it was caused by running multiple replicas without setting the DNS_NODES value...so I think there was some kind of conflict causing it...almost positive...with this fix its been running now for a while without any issues like before
